src/main/java/com/solace/samples/java/snippets/HowToImplementTracingManualInstrumentation.java

```diff
@@ -0,0 +1,182 @@
+                .extract(Context.current(), receivedMessage, getter);
+
+        // Then set the extractedContext as current context
+        try (Scope scope = extractedContext.makeCurrent()) {
+            //...
+        }
+    }
+
+    /**
+     * Example how to inject a tracing context (including span and baggage) in the Solace PubSub+ Java Message
+     * and generate a SEND span for the published message.
+     *
+     * @param message            A Solace PubSub+ Java Message that support tracing context propagation.
+     * @param messagePublisher   Solace PubSub+ Java Message publisher that can publish direct messages
+     * @param messageDestination message destination
+     * @param openTelemetry      The entry-point to telemetry functionality for tracing, metrics and
+     *                           baggage.
+     * @param tracer             Tracer is the interface for Span creation and interaction with the
+     *                           in-process context.
+     */
+    void howToCreateSpanAndBaggageOnMessagePublish(OutboundMessage message, DirectMessagePublisher messagePublisher,
+                                                    Topic messageDestination, OpenTelemetry openTelemetry,
+                                                    Tracer tracer) {
+
+        // Create a new span with a current context as parent of this span
+        final Span sendSpan = tracer
+                .spanBuilder("mySolacePublisherApp" + " " + MessagingOperationValues.PROCESS)
+                .setSpanKind(SpanKind.CLIENT)
+                // published to a topic endpoint (non-temporary)
+                .setAttribute(SemanticAttributes.MESSAGING_DESTINATION_KIND, MessagingDestinationKindValues.TOPIC)
+                .setAttribute(SemanticAttributes.MESSAGING_TEMP_DESTINATION, false)
+                // Set more attributes as needed
+                //.setAttribute(...)
+                //.setAttribute(...)
+                .setParent(Context.current()) // set current context as parent
+                .startSpan();
+
+        // Create baggage object
+        Baggage baggage = BaggageUtil.extractBaggage("key1=val1,key2=val2");
+
+        // set sendSpan as new current context
+        try (Scope scope = sendSpan.makeCurrent()) {
+            // set baggage in the current context
+            try (Scope scope1 = baggage.storeInContext(Context.current()).makeCurrent()) {
+                final SolacePubSubPlusJavaTextMapSetter setter = new SolacePubSubPlusJavaTextMapSetter();
+                final TextMapPropagator propagator = openTelemetry.getPropagators().getTextMapPropagator();
+                // then inject current context with send span and baggage into the message
+                propagator.inject(Context.current(), message, setter);
+                // publish message to the given topic
+                messagePublisher.publish("simple message to the world", messageDestination);
```

Comment:
You pass in an OutboundMessage `message` to this method, inject some context, but the next line `publish()` doesn't send this message... it sends some random string instead.